src/extension-runners/chromium.js

```diff
@@ -114,6 +129,11 @@ export class ChromiumExtensionRunner {
       this.params.extensions.map(({sourceDir}) => sourceDir)
     ).join(',');
 
+    this.params.extensions.forEach(({sourceDir, manifestData}) => {
+      const { name, version } = manifestData;
+      this.extensionsSourceDirs.set(sourceDir, `${name}-${version}`);
```

Comment:
name+version is just one of the ways to identify an extension, and not fully reliable either.
When the developer updates one of those fields in manifest.json, the mapping is going to become obsolete. (another way that the logic could break is when the extension manifest is localized, with different default languages for the browser compared to `web-ext`)

Rather than storing two maps, I would prefer that you maintain one map in a more reliable way.

Have you tried to find all unpacked extensions (`management.getAll`), filtered by `installType == "development"`, and see if the order matches with the order of `--load-extension`? If that works, it would be an easier and a reliable way to map extension IDs.

If you wish, you can also add additional logic to verify that the ID of the installed extension matches with the started-up extension.